returnn/tf/layers/basic.py

```diff
@@ -2637,6 +2640,12 @@ def get_out_data_from_opts(cls, name, axes, sources=(), n_out=NotSpecified, out_
         input_data=input_data, merge_axes=axes, old_axis=input_data.get_batch_axis(old_axis))
       if new_axis != data.batch_dim_axis:
         data.size_placeholder[data.get_batch_axis_excluding_batch(new_axis)] = dyn_size
+    if input_data.batch_dim_axis in axes and data.batch:
+      for axis in axes:
+        if axis != input_data.batch_dim_axis:
+          data.batch = data.batch.copy_extend_with_padded_dim_tag(
```

Comment:
Yes, this is more a case of "just-not-implemented-yet".
But you are right, if someone used this case before, this would break then.
Not sure this is the case.